src/cc/bpf_module.cc

```diff
@@ -95,6 +98,8 @@ class MyMemoryManager : public SectionMemoryManager {
 
 BPFModule::BPFModule(unsigned flags)
     : flags_(flags), ctx_(new LLVMContext) {
+  InitializeNativeTarget();
```

Comment:
The idea is to run a native sscanf function that is tailored to to the key/leaf type. The problem with building a custom parser is that the alignment of the struct is convoluted, but the Type that is kept in the llvm::Module is accurate according to how the frontend (clang or lex, doesn't matter) compiled the struct, taking alignments, padding, and so on into account. This implementation leverages that knowledge, but as a downside uses native JITed code. We can discuss in person the overall approach if this doesn't satisfy. I acknowledge that the dependency is an ugly one.